### PR TITLE
Added question and result page

### PR DESCRIPTION
**Pull request by vrinda-mahajan**
Files to check:-

src/context/globalContext.jsx
src/reducer/globalReducer.jsx
src/pages/questions/questions.jsx
src/components/card/question/questionCard.jsx
src/components/card/question/answer/answer.jsx
src/pages/result/result.jsx

Kindly review the PR.

### CODE REVIEW

File: src/components/card/category/categoryCard.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
+import { useGlobal } from "context/globalContext"
 import { Link } from "react-router-dom"
 
 export const CategoryCard = ({category}) => {
-    const {type,img} = category
+    const {type,img,id} = category
+    const {globalDispatch} = useGlobal()
+
```
Comment (Radheshyam1234) [inline src/components/card/category/categoryCard.jsx:6]: Youe can also destructure like this : 
 export const CategoryCard = ({category:{type,img,id}) =>{}

File: src/reducer/globalReducer.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export const globalInitialState = {
+    quizData: [],
+    answers: {},
+    points: {},
+    score: 0,
+    selectedQuiz: "",
+}
+
+export const globalReducer = (state,action) => {
+    const {type,payload} = action
```
Comment (Radheshyam1234) [inline src/reducer/globalReducer.jsx:11]: same destructring format here also 
globalReducer=(state,{type,payload)}=>{}

**Review by Radheshyam1234 — APPROVED** [review]
Codes look good and well structured.
Go ahead to merge.

File: src/pages/questions/questions.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        const quizColRef = collection(db,"quizzes");
+        const quizCategory = query(quizColRef,where("category","==",selectedQuiz));
+        (async()=>{
+            try {
+                const data = await getDocs(quizCategory);
+                const allQuizzes = data.docs.map(doc=>({id:doc.id,...doc.data()}));
+                globalDispatch({type:"SET_QUIZ_DATA",payload:allQuizzes})
+                setLoading(false)
+            } catch (err) {
+                console.log(err)
```
Comment (sri-rishi) [inline src/pages/questions/questions.jsx]: Suggestion: can use console.error for errors

**Review by sri-rishi — APPROVED** [review]
code quality is good, feel free to merge

**Comment by vrinda-mahajan** [comment]
Thanks @Radheshyam1234  and @sri-rishi for the PR review.

**Pull request merged**